SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc

```diff
@@ -198,44 +240,39 @@ void Phase2TrackerDigitizerAlgorithm::primary_ionization(
       << " " << hit.exitPoint().y() - hit.entryPoint().y() << " " << hit.exitPoint().z() - hit.entryPoint().z() << " "
       << hit.particleType() << " " << hit.pabs();
 
-  std::vector<float> elossVector;  // Eloss vector
+  std::vector<float> elossVector;
   elossVector.reserve(NumberOfSegments);
   if (fluctuateCharge_) {
     // Generate fluctuated charge points
-    fluctuateEloss(hit.particleType(), hit.pabs(), eLoss, length, NumberOfSegments, elossVector);
+    elossVector = fluctuateEloss(hit.particleType(), hit.pabs(), eLoss, length, NumberOfSegments);
+  } else {
+    float averageEloss = eLoss / NumberOfSegments;
+    std::fill(std::begin(elossVector), std::end(elossVector), averageEloss);
```

Comment:
`reserve()` doesn't actually create entries, so I don't think this `fill()` will work. Rather:
```
elossVector.resize(NumberOfSegments, averageEloss);
```